### PR TITLE
feat: Make create value endpoint fully support JSON-LD (DEV-4305)

### PR DESCRIPTION
**Pull request by seakayone**
- **Introduce model for KnoraIris**
-- **Introduce ValueIri**
-- **Introduce ResourceIri**
-- **Introduce ResourceClassIri**
-- **Introduce PropertyIri**

- **Introduce `KnoraApiCreateValueModel` backed by an rdf resource**

- **Move jena related code to dedicated `...Ops` extension methods**

- **Replace parsing from `JsonLdDocument` with rdf parsing for `CreateValueContentV2`**
- **Inline `ValuesResponderV2` interface**
- **Remove unused `requestingUser` from messages**

### Task Description/Number

Issue Number: DEV-

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [x] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [ ] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)
- [ ] deprecated: Deprecation warning (ideally referencing a migration guide)

### Basic requirements for bug fixes and features

- [x] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### CODE REVIEW

File: integration/src/test/scala/org/knora/webapi/E2ESpec.scala [hunk 1]
```diff
@@ -120,7 +120,7 @@ abstract class E2ESpec
   protected def singleAwaitingRequest(
     request: HttpRequest,
     timeout: Option[zio.Duration] = None,
-    printFailure: Boolean = false,
+    printFailure: Boolean = true,
```
Comment (mpro7) [inline integration/src/test/scala/org/knora/webapi/E2ESpec.scala:123]: Just to make sure, this is desired change, not forgotten in cleanup?

**Review by mpro7 — APPROVED** [review]
Looks good, thanks! Would it make sense to add short documentation in Scaldoc to core Classes/Objects/Traits?

File: integration/src/test/scala/org/knora/webapi/E2ESpec.scala (hunk 1 above)
Comment (seakayone) [reply integration/src/test/scala/org/knora/webapi/E2ESpec.scala:123]: I am turning this on anyway any time I try to understand why the tests fails. I think it is also helpful on CI.

File: webapi/src/test/scala/org/knora/webapi/slice/common/KnoraApiValueModelSpec.scala [hunk 2]
```diff
@@ -58,17 +111,520 @@ object KnoraApiValueModelSpec extends ZIOSpecDefault {
...
         for {
-          model            <- KnoraApiValueModel.fromJsonLd(json)
-          resourceClassIri <- model.rootResourceClassIri
-        } yield assertTrue(resourceClassIri.toString == "http://0.0.0.0:3333/ontology/0001/anything/v2#Thing")
+          model <- KnoraApiCreateValueModel.fromJsonLd(json)
+        } yield assertTrue(model.resourceClassIri.toString == "http://0.0.0.0:3333/ontology/0001/anything/v2#Thing")
       }
     },
-  ).provideSome[Scope](IriConverter.layer, StringFormatter.test)
+    test("valueNode properties should be present") {
```
Comment (BalduinLandolt) [inline webapi/src/test/scala/org/knora/webapi/slice/common/KnoraApiValueModelSpec.scala:125]: remark (non-blocking): I'm not really clear what is meant by "valueNode". Probably it's just me not being deeply enough in the matter, but maybe you find a more descriptive phrasing? Or you can explain it to me? :)

File: webapi/src/test/scala/org/knora/webapi/slice/common/KnoraApiValueModelSpec.scala [hunk 3]
```diff
@@ -58,17 +111,520 @@ object KnoraApiValueModelSpec extends ZIOSpecDefault {
...
+      } yield assertTrue(content == BooleanValueContentV2(ApiV2Complex, true, None))
+    },
+    test("should parse GeomValueContentV2") {
+      for {
+        model <- KnoraApiCreateValueModel.fromJsonLd("""{
+                                                       |  "@id": "http://rdfh.ch/0001/a-thing",
+                                                       |  "@type": "ex:Thing",
+                                                       |  "ex:someGeom": {
+                                                       |    "@type": "ka:GeomValue",
+                                                       |    "ka:geometryValueAsGeometry": "{}"
```
Comment (BalduinLandolt) [inline webapi/src/test/scala/org/knora/webapi/slice/common/KnoraApiValueModelSpec.scala:208]: remark (non-blocking): the geometryValueAsGeometry `"{}"` seems suboptimal to me. I guess it's a valid value, but it's definitely not a good illustration of what these values should look like in reality, if we consider tests as "codified documentation".  
I don't want to clutter your test with something that is actually beside the point of the test. But I kind of like the idea of having meaningful values in these tests. How do you feel about this?

File: webapi/src/test/scala/org/knora/webapi/slice/common/KnoraApiValueModelSpec.scala [hunk 4]
```diff
@@ -58,17 +111,520 @@ object KnoraApiValueModelSpec extends ZIOSpecDefault {
...
+        model <-
+          KnoraApiCreateValueModel.fromJsonLd(
+            s"""
+               |{
+               |  "@id" : "http://rdfh.ch/0001/a-thing",
+               |  "@type" : "ex:Thing",
+               |  "ex:hasOtherThingValue" : {
+               |    "@id" : "http://rdfh.ch/0001/a-thing/values/mr9i2aUUJolv64V_9hYdTw",
+               |    "@type" : "ka:GeonameValue",
+               |    "ka:geonameValueAsGeonameCode" : "foo"
```
Comment (BalduinLandolt) [inline webapi/src/test/scala/org/knora/webapi/slice/common/KnoraApiValueModelSpec.scala:335]: again, maybe a realistic value would be nicer? Proposal: `"2661604"` (Basel)

File: webapi/src/test/scala/org/knora/webapi/slice/common/KnoraApiValueModelSpec.scala [hunk 5]
```diff
@@ -58,17 +111,520 @@ object KnoraApiValueModelSpec extends ZIOSpecDefault {
...
+        model <-
+          KnoraApiCreateValueModel.fromJsonLd(
+            s"""
+               |{
+               |  "@id" : "http://rdfh.ch/0001/a-thing",
+               |  "@type" : "ex:Thing",
+               |  "ex:hasOtherThingValue" : {
+               |    "@id" : "http://rdfh.ch/0001/a-thing/values/mr9i2aUUJolv64V_9hYdTw",
+               |    "@type" : "ka:ColorValue",
+               |    "ka:colorValueAsColor" : "red"
```
Comment (BalduinLandolt) [inline webapi/src/test/scala/org/knora/webapi/slice/common/KnoraApiValueModelSpec.scala:358]:
```suggestion
               |    "ka:colorValueAsColor" : "#ff0000"
```

**Review by BalduinLandolt — APPROVED** [review]
I didn't see anything alarming, but due to the nature of the PR, these wouldn't be easy to spot. But with the test coverage we have, I feel confident that this can be merged.  
I added some small remarks, but nothing important. Feel free to ignore them or address them in the upcoming follow-up PR so this one can be merged sooner.

**Comment by seakayone** [comment]
@BalduinLandolt I do understand for your desire of having more "realistic" values in the tests. My goal was to have minimal test data in the newly created tests. The other more complicated cases are covered by the existing tests. I leave these values in the test as to demonstrate that these are valid values. 

Should you think that those values are not valid we may want to follow up on that.

File: webapi/src/test/scala/org/knora/webapi/slice/common/KnoraApiValueModelSpec.scala (hunk 2 above)
Comment (seakayone) [reply webapi/src/test/scala/org/knora/webapi/slice/common/KnoraApiValueModelSpec.scala:125]: I will clarify the title of this test in the other PR. `valueNode` does not have a special meaning anymore the anyway.

Comment (seakayone) [reply webapi/src/test/scala/org/knora/webapi/slice/common/KnoraApiValueModelSpec.scala:125]: Here is the clarification: https://github.com/dasch-swiss/dsp-api/pull/3413/commits/035efb606f2b1e12d7f755f616d7fcb0d1ca184a

Comment (BalduinLandolt) [reply webapi/src/test/scala/org/knora/webapi/slice/common/KnoraApiValueModelSpec.scala:125]: thanks!

**Comment by BalduinLandolt** [comment]
> @BalduinLandolt I do understand for your desire of having more "realistic" values in the tests. My goal was to have minimal test data in the newly created tests. The other more complicated cases are covered by the existing tests. I leave these values in the test as to demonstrate that these are valid values.
> 
> Should you think that those values are not valid we may want to follow up on that.

Fine by me, I'm not even sure we need to follow up on it. Thanks

**Pull request merged**